compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -793,6 +793,13 @@ typeOf' = \case
   EUpdate upd -> typeOfUpdate upd
   EScenario scen -> typeOfScenario scen
   ELocation _ expr -> typeOf' expr
+  EViewInterface iface template viewtype expr -> do
+    checkImplements template iface
+    iface <- inWorld (lookupInterface iface)
+    unless (alphaType (intView iface) viewtype) $
+      throwWithContext ETypeMismatch{foundType = viewtype, expectedType = intView iface, expr = Nothing}
+    checkExpr expr (TCon template)
```

Comment:
I'm not sure what type we want for the argument here, but `ECallInterface` takes a `TCon iface`

```suggestion
    checkExpr expr (TCon iface)
```

@remyhaemmerle-da what do you think?

Reply:
The `view` construct should be applied to an interface at a point we do not know the actual type of the template. 
As explained in my previous comment, I think the definition of `EViewInterface` should be something like.

```
EViewInterface
    { viewInterfaceInterface :: !(Qualified TypeConName)
    , viewInterfaceExpr :: !Expr
    }
```